test/src/test/resources/fixtures/new hydration/new batching/new-batching-absent-source-input.yml

```diff
@@ -0,0 +1,215 @@
+name: "new batching absent source input"
+enabled: true
+# language=GraphQL
+overallSchema:
+  activity: |
+    type Query {
+      activity: [Activity]
+    }
+    union ActivityContent = Issue | Comment
+    type Activity {
+      id: ID!
+      contentId: ID
+      content: ActivityContent
+        @hydrated(
+          service: "comments"
+          field: "commentsByIds"
+          arguments: [
+            {name: "ids" value: "$source.contentId"}
+          ]
+        )
+        @hydrated(
+          service: "issues"
+          field: "issuesByIds"
+          arguments: [
+            {name: "ids" value: "$source.contentId"}
+          ]
+        )
+    }
+  # language=GraphQL
+  issues: |
+    type Query {
+      issuesByIds(ids: [ID!]!): [Issue!]
+    }
+    type Issue {
+      id: ID!
+      title: String
+    }
+  # language=GraphQL
+  comments: |
+    type Query {
+      commentsByIds(ids: [ID!]!): [Comment!]
+    }
+    type Comment {
+      id: ID!
+      content: String
+    }
+# language=GraphQL
+underlyingSchema:
+  # language=GraphQL
+  activity: |
+    type Query {
+      activity: [Activity]
+    }
+    type Activity {
+      id: ID!
+      contentId: ID
+    }
+  # language=GraphQL
+  issues: |
+    type Query {
+      issuesByIds(ids: [ID!]!): [Issue!]
+    }
+    type Issue {
+      id: ID!
+      title: String
+    }
+  # language=GraphQL
+  comments: |
+    type Query {
+      commentsByIds(ids: [ID!]!): [Comment!]
+    }
+    type Comment {
+      id: ID!
+      content: String
+    }
+# language=GraphQL
+query: |
+  {
+    activity {
+      content {
+        __typename
+        ... on Issue {
+          id
+          title
+        }
+        ... on Comment {
+          id
+          content
+        }
+      }
+    }
+  }
+variables: { }
+serviceCalls:
+  - serviceName: "activity"
+    request:
+      # language=GraphQL
+      query: |
+        {
+          activity {
+            __typename__batch_hydration__content: __typename
+            batch_hydration__content__contentId: contentId
+            batch_hydration__content__contentId: contentId
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "activity": [
+            {
+              "__typename__batch_hydration__content": "Activity"
```

Comment:
that is not a legal answer, is it? I mean there is no fragment/type selector involved and therefore every key must be present.

Reply:
Yeah I was creating tests with different theories with how the bug was occurring.

I just left it in anyway. I don't think there's much harm.

(I eventually just pulled in the data from the logs and got an exact replication).